_deploy/r/gnoswap/common/halt.gno

```diff
@@ -39,34 +55,22 @@ func SetHaltByAdmin(halt bool) {
 			),
 		))
 	}
-
 	setHalt(halt)
-
-	prevAddr, prevRealm := getPrev()
-	if halt {
-		std.Emit(
-			"SetHaltByAdmin",
-			"prevAddr", prevAddr,
-			"prevRealm", prevRealm,
-			"halt", ufmt.Sprintf("%t", halt),
-		)
-	} else {
-		std.Emit(
-			"UnsetHaltByAdmin",
-			"prevAddr", prevAddr,
-			"prevRealm", prevRealm,
-			"halt", ufmt.Sprintf("%t", halt),
-		)
-	}
 }
 
-// SetHalt sets the halt status.
-// Only governance contract can execute this function via proposal
+// SetHalt allows the governance contract to set the halt status of the system.
+// Only the governance contract can execute this function through a proposal process.
+// If the caller is not the governance contract, the function panics with an errNoPermission error.
```

Comment:
```suggestion
// If the caller is not the governance contract, the function panics with an errNoPermission error.
```

IMHO, it can be removed. Line 69 explains same thing.